js/bloodmallet_standard.js

```diff
@@ -968,41 +973,20 @@ function update_chart() {
+  const baselineDps=trinketArray["trinkets"]["baseline"][minIlvl];
 
-  for (itemlevel_position in loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"]) {
+  for (let itemlevel_position in trinketArray["Simulated itemlevels"]) {
 
-    var itemlevel = loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"][itemlevel_position];
+    var itemlevel = trinketArray["Simulated itemlevels"][itemlevel_position];
     var itemlevel_dps_values = [];
 
     for (trinket of dps_ordered_trinkets) {
+      const currentItemDps=trinketArray["trinkets"][trinket][itemlevel];
+      const lastItemDps=trinketArray["trinkets"][trinket][trinketArray["Simulated itemlevels"][String(Number(itemlevel_position) + 1)]] || baselineDps;
 
-      // check for zero dps values and don't change them
-      if (Number(loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][itemlevel]) > 0) {
-
-        // if lowest itemlevel is looked at, substract baseline
-        if (itemlevel_position == loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"].length - 1) {
-
-          itemlevel_dps_values.push(loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][itemlevel] - loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"]["baseline"][Math.min(...loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"])]);
-
-
-        } else { // else substract lower itemlevel value of same item
-
-          // if lower itemlevel is zero we have to assume that this item needs to be compared now to the baseline
-          if (loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"][String(Number(itemlevel_position) + 1)]] == 0) {
 
-            itemlevel_dps_values.push(loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][itemlevel] - loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"]["baseline"][Math.min(...loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"])]);
-
-          } else { // standard case, next itemlevel is not zero and can be used to substract from the current value
-
-            itemlevel_dps_values.push(loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][itemlevel] - loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][loaded_data[chosen_class][chosen_spec][data_view][fight_style]["Simulated itemlevels"][String(Number(itemlevel_position) + 1)]]);
-          }
-
-        }
-
-      } else {
-
-        itemlevel_dps_values.push(loaded_data[chosen_class][chosen_spec][data_view][fight_style]["trinkets"][trinket][itemlevel]);
-      }
+      itemlevel_dps_values.push((currentItemDps - lastItemDps)||0);
```

Comment:
Needs a check for if currentItemDps == 0, in that case only push 0 to prevent this: 
![image](https://user-images.githubusercontent.com/9075987/40019716-521a9768-57c0-11e8-8f38-4c179d5993da.png)